### ISSUE TITLE
Compiling Error with FlowAccumulation 

### ISSUE THREAD

**Issue by tcalderon-space** [landlab/landlab#2076]
Hello, I was going over the tutorial for "What makes up a Landlab Model"

I am able to import FlowAccumulator however when I try to use it I run in to a field error.

Below is the error I run into.
I tried using another tutorial that called flowaccumulator and still ran into the same issue...

"                           Traceback (most recent call last)
Cell In[25], line 3
      1 mg = RasterModelGrid((10, 10), xy_spacing=(1.0, 2.0))  # 100 nodes, dy=1., dx=2. # also wrong
----> 3 fa = FlowAccumulator(mg)

File ~\anaconda3\Lib\site-packages\landlab\components\flow_accum\flow_accumulator.py:695, in FlowAccumulator.__init__(self, grid, surface, flow_director, runoff_rate, depression_finder, **kwargs)
    693 # save elevations and node_cell_area to class properites.
    694 self._surface = surface
--> 695 self._surface_values = return_array_at_node(grid, surface)
    697 if self._is_Network:
    698     try:

File ~\anaconda3\Lib\site-packages\landlab\utils\decorators.py:318, in use_field_name_array_or_value.__call__.<locals>._wrapped(grid, vals, *args, **kwds)
    316         vals = grid[self._at][vals]
    317     else:
--> 318         raise FieldError(vals)
    319 else:
    320     expected_size = grid.size(self._at)

FieldError: topographic__elevation"

If I could get any feedback that would be great. Thank you!

**Comment by mcflugen** [comment]
@tcalderon-space Could you please provide a link to the tutorial you're referring to?

**Comment by tcalderon-space** [comment]
Yes, its https://landlab.readthedocs.io/en/v2.9.2/user_guide/build_a_model.html

**Comment by mcflugen** [comment]
@tcalderon-space Wow, that example is way out-of-date (and contains typos and mistakes). Thanks for reporting!

I'll open a new pull request with updates to fix the example.

The particular error you are getting is because your grid is missing a field (`"topographic__elevation"`), which the `FlowAccumulator` is requires.

**Comment by tcalderon-space** [comment]
Ok, thank you. I tried adding topography parameter, but it still didn't work.. I am wondering if there is something else that is causing it?

**Comment by mcflugen** [comment]
How did you add the topography parameter? There are other ways to do it, but the following works for me,
```python
from landlab import RasterModelGrid
from landlab.components import FlowAccumulator

grid = RasterModelGrid((10, 10), xy_spacing=(1.0, 2.0))
grid.add_zeros("topographic__elevation", at="node")
flow_accumulator = FlowAccumulator(grid)
```

**Comment by mcflugen** [comment]
The `surface` keyword simply tells the flow accumulator the name of the field over which to accumulate flow. It doesn't create a new field.

**Comment by tcalderon-space** [comment]
Hello, would it be possible to be able to contribute to updating the tutorial page?  I was able to make adjustments with the flowaccumulation and also noticed that there was an error with PrecipitationDistribution. (I'll open another issue for this)
